README.md

```diff
@@ -32,30 +32,46 @@ Here is the high-level diagram comparing a traditional architecture vs a 3factor
 
 ### Stack
 
-NodeJS 8.1
+NodeJS 8.1+
 
 Postgres 9.5+
 
 Hasura GraphQL Engine
 
-AWS Lambda
+AWS Lambda (optional, for hosting serverless functions)
 
-### Step 1: Model application
+### Step 1: Setup a realtime GraphQL interface
 
-The core modeling of a 3factor app is very similar to traditional application modeling: we start by defining the schema and then define different functional components. The main difference is that 3factor emphasizes that each function be atomic (i.e. either happens completely or doesn't happen at all) and invoked via (persisted) events.
+#### Hasura
 
-#### Schema
+3factor requires the frontend use GraphQL for querying and performing actions. The reason for this is two-fold: fast iteration and realtime feedback. 
 
-The schema of our application is defined in [schema.sql](schema.sql) which you can apply on your postgres database.
+We will use Hasura to get GraphQL APIs over our existing Postgres database. We will use Docker to run Hasura. If you do not have Docker, you can install it from [here](https://store.docker.com/search?type=edition&offering=community).
 
+Run the following command:
 ```bash
+$ docker-compose up -d
+```
 
-$ export POSTGRES_CONNECTION_STRING='postgres://postgres:password@localhost:5432/postgres'
+#### Schema
 
-$ psql $POSTGRES_CONNECTION_STRING < schema.sql
+The schema of our application is defined in [schema.sql](schema.sql) which you can apply on your Postgres database.
 
+```bash
+$ docker exec -i 3factor_postgres psql -U postgres postgres < schema.sql
```

Comment:
The target container name should be `postgres`, i.e.
```
$ docker exec -i postgres psql -U postgres postgres < schema.sql
```
instead in order to match the container name defined in `docker-compose.yaml`.